### PR TITLE
Feature search

### PR DESCRIPTION
**Pull request by khihadysucahyo**
## Changes
- init elasticsearch & search module
- paginate elastic response
- elastic seacrh aggregation
- meta aggs proto on endpoint search (still hardcoded)

## Todo
- FIXME: meta aggs proto on endpoint search

cc @jabardigitalservice/jds-backend 

## Evidence
project: Portal Jabar
title: init elasticsearch & search module
participants: @khihadysucahyo @rachadiannovansyah @ayocodingit @firmanJS

### CODE REVIEW

File: core-service/src/modules/search/repository/elastic/elastic_search.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+// type alias for map query
+type q map[string]interface{}
+
+func (es *elasticSearchRepository) Fetch(ctx context.Context, params *domain.Request) (docs []domain.SearchListResponse, total int64, aggs interface{}, err error) {
+
+	var buf bytes.Buffer
+	query := q{
+		"query": q{
+			"multi_match": q{
```
Comment (ayocodingit) [inline core-service/src/modules/search/repository/elastic/elastic_search.go:60]: Untuk metode elastic, Jadinya pake metode multi match yah @khihadysucahyo ?

Just Ask 
untuk aggs nya untuk semua metode apakah sama aja atau berbeda ?

**Review by ayocodingit — APPROVED** [review]

File: core-service/src/modules/search/delivery/http/search_handler.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+// FetchNews will fetch the content based on given params
+func (h *SearchHandler) FetchSearch(c echo.Context) error {
+	ctx := c.Request().Context()
+	params := helpers.GetRequestParams(c)
+	listSearch, tot, aggs, err := h.SUsecase.Fetch(ctx, &params)
+	if err != nil {
+		return err
+	}
+	res := helpers.Paginate(c, listSearch, tot, params)
```
Comment (rachadiannovansyah) [inline core-service/src/modules/search/delivery/http/search_handler.go:32]: letsgoo but it's seems still `tot` out there 🥲

**Review by rachadiannovansyah — APPROVED** [review]

**Review by firmanJS — APPROVED** [review]

File: core-service/src/modules/search/repository/elastic/elastic_search.go (hunk 1 above)
Comment (khihadysucahyo) [reply core-service/src/modules/search/repository/elastic/elastic_search.go:60]: metode paan nich

Comment (khihadysucahyo) [reply core-service/src/modules/search/repository/elastic/elastic_search.go:60]: saat ini basic multi match ke field content dan title

Comment (khihadysucahyo) [reply core-service/src/modules/search/repository/elastic/elastic_search.go:60]: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html

Comment (ayocodingit) [reply core-service/src/modules/search/repository/elastic/elastic_search.go:60]: > metode paan nich

Yang waktu diskusi ada beberapa metode di elastic

Comment (khihadysucahyo) [reply core-service/src/modules/search/repository/elastic/elastic_search.go:60]: masi pake yg default https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html#type-best-fields


bahasa inggris woooyy

File: core-service/src/modules/search/delivery/http/search_handler.go (hunk 2 above)
Comment (khihadysucahyo) [reply core-service/src/modules/search/delivery/http/search_handler.go:32]: IYA IH MASIH TOT
next pr deh.

**Pull request merged**